wpiutil/src/main/java/edu/wpi/first/util/cleanup/CleanupPool.java

```diff
@@ -0,0 +1,50 @@
+package edu.wpi.first.util.cleanup;
+
+import java.util.Stack;
+
+import edu.wpi.first.util.ErrorMessages;
+
+/**
+ * An object containing a Stack of AutoCloseable objects that are closed
+ * when this object is closed.
+ */
+public class CleanupPool implements AutoCloseable {
+
+  private final Stack<AutoCloseable> m_closers = new Stack<AutoCloseable>();
```

Comment:
Do we want to use weak refs here so this doesn't prevent GC? Though the scope this should be used in is rather different.

Reply:
No. You're right the scope here is much different, and theres no need for the overhead of a weak ref.